Can't compile for server on FreeBSD

Godot version: latest source
OS version: FreeBSD 10.3-RELEASE-p3

When compiling for server (platform=server) on FreeBSD compilation fails at very beginning stage.
Error message:

```
scons: Building targets ...
g++ -o platform/server/godot_server.server.tools.64.o -c -g2 -Wall -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -DFREETYPE_ENABLED -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE -DENABLE_DEPRECATED -DSERVER_ENABLED -DUNIX_ENABLED -DMUSEPACK_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DOPUS_ENABLED -DTHEORA_ENABLED -DTHEORALIB_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/server -Idrivers/freetype -Idrivers/freetype/freetype/include -Idrivers/freetype/include platform/server/godot_server.cpp
In file included from core/typedefs.h:37:0,
                 from ./main/main.h:37,
                 from platform/server/godot_server.cpp:29:
platform/server/platform_config.h:29:20: fatal error: alloca.h: No such file or directory
 #include <alloca.h>
                    ^
compilation terminated.
scons: *** [platform/server/godot_server.server.tools.64.o] Error 1
scons: building terminated because of errors.

```

On FreeBSD terminal clone repository, enter it and type:

> scons platform=server

COMMENT:
I was able to fix above error by adding:

```
#ifdef __linux__
#include <alloca.h>
#endif
#if defined(__FreeBSD__) || defined(__OpenBSD__)
#include <stdlib.h>
#define PTHREAD_BSD_SET_NAME
#endif

```

to the 

> platform/server/platform_config.h

but then another has error appeared:

> g++ -o modules/enet/register_types.server.tools.64.o -c -g2 -Wall -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -DFREETYPE_ENABLED -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE -DENABLE_DEPRECATED -DSERVER_ENABLED -DUNIX_ENABLED -DMUSEPACK_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DOPUS_ENABLED -DTHEORA_ENABLED -DTHEORALIB_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/server -Idrivers/freetype -Idrivers/freetype/freetype/include -Idrivers/freetype/include modules/enet/register_types.cpp
> In file included from modules/enet/networked_multiplayer_enet.h:5:0,
>                  from modules/enet/register_types.cpp:31:
> modules/enet/enet/enet.h:18:23: fatal error: enet/unix.h: No such file or directory
>  #include "enet/unix.h"
>                        ^
> compilation terminated.
> scons: **\* [modules/enet/register_types.server.tools.64.o] Error 1

--- a/platform/server/detect.py
+++ b/platform/server/detect.py
@@ -1,4 +1,5 @@
 import os
+import platform
 import sys
 
 
@@ -132,7 +133,12 @@ def configure(env):
     env.Append(CPPPATH=['#platform/server'])
     env.Append(CPPFLAGS=['-DSERVER_ENABLED', '-DUNIX_ENABLED'])
     env.Append(LIBS=['pthread'])
-    env.Append(LIBS=['dl'])
+
+    if (platform.system() == "Linux"):
+        env.Append(LIBS=['dl'])
+
+    if (platform.system().find("BSD") >= 0):
+        env.Append(LIBS=['execinfo'])
 
     # Link those statically for portability
     if env['use_static_cpp']:

--- a/platform/server/platform_config.h
+++ b/platform/server/platform_config.h
@@ -28,4 +28,10 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
+#ifdef __linux__
 #include <alloca.h>
+#endif
+#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#include <stdlib.h>
+#define PTHREAD_BSD_SET_NAME
+#endif

--- a/platform/x11/crash_handler_x11.cpp
+++ b/platform/x11/crash_handler_x11.cpp
@@ -32,8 +32,9 @@
 #define CRASH_HANDLER_ENABLED 1
 #endif
 
+#include "crash_handler_x11.h"
 #include "main/main.h"
-#include "os_x11.h"
+#include "os/os.h"
 #include "project_settings.h"
 
 #ifdef CRASH_HANDLER_ENABLED

